### ISSUE TITLE
Inquiry About Parallel Computing

### ISSUE THREAD

**Issue by lidafa0206** [JohannesBuchner/UltraNest#150]
Thanks for the great package!

I hope this message finds you well. I have a question regarding parallel computing. When I run the command `mpiexec -np 4 xxx.py`, I notice that I receive four separate results. Does this imply that MPI is executing four independent calculations on the model simultaneously, rather than enhancing the computation speed by a factor of four?

Given that my program involves high-dimensional parameters and tends to run slowly, I would greatly appreciate any advice on methods to accelerate its execution.

Thank you very much for your time and assistance.

**Comment by JohannesBuchner** [comment]
yes, please have a look at https://johannesbuchner.github.io/UltraNest/debugging.html#Parallelisation-issues

**Comment by JohannesBuchner** [comment]
Hope you were able to debug it in the meanwhile, if not, please reopen.
